File: app/src/main/java/com/centaury/cataloguemovie/MovieCatalogueApp.java
package com.centaury.cataloguemovie;

import android.app.Activity;
import android.app.Application;

import com.androidnetworking.AndroidNetworking;
import com.centaury.cataloguemovie.di.AppInjector;

import javax.inject.Inject;

import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasActivityInjector;

/**
 * Created by Centaury on 10/24/2019.
 */
public class MovieCatalogueApp extends Application implements HasActivityInjector {

    @Inject
    DispatchingAndroidInjector<Activity> dispatchingAndroidInjector;

    @Override
    public void onCreate() {
        super.onCreate();

        AppInjector.init(this);

        AndroidNetworking.initialize(getApplicationContext());
    }

    @Override
    public AndroidInjector<Activity> activityInjector() {
        return dispatchingAndroidInjector;
    }
}

File: app/src/main/kotlin/com/centaury/cataloguemovie/MovieCatalogueApp.kt
package com.centaury.cataloguemovie

import android.app.Application
import com.centaury.cataloguemovie.di.component.AppComponent
import dagger.hilt.EntryPoints
import dagger.hilt.android.HiltAndroidApp
import timber.log.Timber

/**
 * Created by Centaury on 10/24/2019.
 */
@HiltAndroidApp
class MovieCatalogueApp : Application() {

    val appComponent: AppComponent by lazy {
        EntryPoints.get(this, AppComponent::class.java)
    }

    override fun onCreate() {
        super.onCreate()

        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }
    }
}